heron/common/src/java/com/twitter/heron/common/network/SocketChannelHelper.java

```diff
@@ -134,7 +135,7 @@ public boolean sendPacket(OutgoingPacket outgoingPacket) {
     // 1. We spent too much time
     // 2. We have read large enough data
     while ((System.nanoTime() - startOfCycle - readReadBatchTimeInNs) < 0
-        && (bytesRead < readBatchSizeInBytes)) {
+        && (bytesRead < readBatchSize.asBytes())) {
```

Comment:
like other questions, shall we also do `ByteAmount` here? I assume you are following some principle here: what are converted to ByteAmount, what are converted. Can you elaborate a bit?